rootfs/scripts/utils.sh

```diff
@@ -61,28 +61,157 @@ enqueue() {
+			if [ -n "$__cleanup_timestamp" ] && [ $((__current_timestamp - __cleanup_timestamp)) -gt 300 ]; then
+				echo "[$__ticket_id] Found stale cleanup lock (>5 minutes old), assuming previous cleanup job failed. Taking over cleanup..."
+				# Continue with cleanup (don't return)
+			else
+				echo "[$__ticket_id] Another process is actively cleaning up. We will not cleanup ourselves..."
+				sleep 11
+				return
+			fi
+		else
+			echo "[$__ticket_id] No active cleanup lock found despite push failure. Retrying..."
+			sleep 5
+			# Retry cleanup
+			cleanup_mutex_branch $__branch $__queue_file $__ticket_id
+			return
+		fi
+	fi
+
+	echo "[$__ticket_id] Acquired cleanup lock, waiting 10 seconds for others to observe the cleanup lock"
+	sleep 10
+	echo "[$__ticket_id] Proceeding with cleanup"
+
+	# Create a new orphan branch to replace the current one
+	git switch --orphan gh-action-mutex/temp-branch-$(date +%s) --quiet
+
+	# Create an empty queue file
+	touch $__queue_file
+
+	git add $__queue_file
+	git commit -m "[$__ticket_id] Reset mutex branch" --quiet
+
+	# Force push to replace the branch
+	git push -f --set-upstream origin HEAD:$__branch --quiet
+
+	# Checkout the branch again
+	git checkout $__branch --quiet || git switch --orphan $__branch --quiet
+
+	echo "[$__ticket_id] Mutex branch cleanup completed"
+}
+
 # Wait for the lock to become available
 # args:
 #   $1: branch
 #   $2: queue_file
 #   $3: ticket_id
+#   $4: start_time (optional, used for timeout calculation)
 wait_for_lock() {
 	__branch=$1
 	__queue_file=$2
 	__ticket_id=$3
+	__start_time=${4:-$(date +%s)}
+	__current_time=$(date +%s)
+	__elapsed_time=$((__current_time - __start_time))
+
+	# Check if we've exceeded the timeout
+	if [ -n "$ARG_TIMEOUT" ] && [ $__elapsed_time -ge $ARG_TIMEOUT ]; then
+		echo "[$__ticket_id] Timeout reached after $__elapsed_time seconds"
+
+		if [ "$ARG_CLEANUP_MUTEX_ON_TIMEOUT" = "true" ]; then
+			echo "[$__ticket_id] cleanup-mutex-on-timeout is enabled, cleaning up mutex branch and retrying"
+			cleanup_mutex_branch $__branch $__queue_file $__ticket_id
+			# Re-enqueue ourselves after cleanup
+			enqueue $__branch $__queue_file $__ticket_id
+			# Reset the start time for a new attempt
+			wait_for_lock $__branch $__queue_file $__ticket_id $(date +%s)
+			return
+		else
+			echo "[$__ticket_id] cleanup-mutex-on-timeout is disabled, exiting with error"
+			exit_failure_with_information $__branch
+		fi
+	fi
 
 	update_branch $__branch
 
+	# Check if cleanup is in progress
+	if grep -q "^CLEANUP_IN_PROGRESS:" "$__queue_file" 2>/dev/null; then
+		echo "[$__ticket_id] Cleanup in progress detected, resetting timeout counter"
+		echo "[$__ticket_id] Waiting for cleanup to finish..."
+		sleep 11
+		# Reset the start time to give the cleanup process time to complete
+		wait_for_lock $__branch $__queue_file $__ticket_id $(date +%s)
+		return
+	fi
+
 	# if we are not the first in line, spin
```

Comment:
Ik zou deze comment 3 regels naar beneden schuiven, zonder de if.

Reply:
Misschien hier: `# check if we are at the top of the lockfile, and if it's empty, requeue ourselves`.